OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -96,38 +111,73 @@ public override Activity Tick(Actor self)
 			if (useLastVisibleTarget && !lastVisibleTarget.IsValidFor(self))
 				return NextActivity;
 
-			var pos = self.CenterPosition;
-			var checkTarget = useLastVisibleTarget ? lastVisibleTarget : target;
-
-			if (!soundPlayed && aircraft.Info.TakeoffSounds.Length > 0 && self.IsAtGroundLevel())
-			{
-				Game.Sound.Play(SoundType.World, aircraft.Info.TakeoffSounds, self.World, aircraft.CenterPosition);
-				soundPlayed = true;
-			}
-
 			// We are taking off, so remove influence in ground cells.
 			if (self.IsAtGroundLevel())
+			{
+				if (!soundPlayed && aircraft.Info.TakeoffSounds.Length > 0)
+				{
+					Game.Sound.Play(SoundType.World, aircraft.Info.TakeoffSounds, self.World, aircraft.CenterPosition);
+					soundPlayed = true;
+				}
+
 				aircraft.RemoveInfluence();
+				isTakeOff = true;
+			}
+
+			// If we're a VTOL, rise before flying forward
```

Comment:
IMO we should allow `CanHover` units to turn while climbing.

With my suggestion above, we could call `FlyTowards` with `MovementType.Vertical | MovementType.Turn` for VTOL+Hover vs just `MovementType.Vertical` for non-hovering (unless we want it there too?)

Reply:
> unless we want it there too?

I went for 'yes', as it makes the code a bit simpler, too.